File: 1-50/problem5/lcm.cpp
#include <iostream>
using namespace std;

int gcd(int a, int b) {
  if (a == 0) return b;
  if (b == 0) return a;

  while (b != 0) {
    int temp = b;
    b = a%b;
    a = temp;
  }
  return a;
}

int lcm(int a, int b) {
  return a/gcd(a,b)*b;
}

int main(int argc, char** argv) {
  int l = 1;
  for (int i = 2; i <= 20; i++)
    l = lcm(l, i);

  cout << l << endl;
  return 0;
}

File: 1-50/problem5/lcm.rb
def gcd(a, b)
  return b if a.zero?
  
  while !b.zero?
    a, b = b, a % b
  end
  a
end

def lcm(a, b)
  a/gcd(a, b)*b
end

r = 1
2.upto(20) do |i|
  r = lcm(i, r)
  puts r
end
puts r